sinabs/layers/lif.py

```diff
@@ -163,11 +163,18 @@ def forward(self, input_data: torch.Tensor) -> torch.Tensor:
         """
         batch_size, time_steps, *trailing_dim = input_data.shape
 
-        # Ensure the neuron state are initialized
-        if not self.is_state_initialised() or not self.state_has_shape(
-            (batch_size, *trailing_dim)
-        ):
+        # Ensure the state is initialized.
+        if not self.is_state_initialised():
             self.init_state_with_shape((batch_size, *trailing_dim))
+        
+        if not self.state_has_shape((batch_size, *trailing_dim)):
+            # If the trailing dim has changed, we reinitialize the states.
+            if not self.has_trailing_dimension(trailing_dim):
+                self.init_state_with_shape((batch_size, *trailing_dim))
+            # Otherwise only the batch size has changed.
```

Comment:
this now basically handles the edge case for the last mini batch, which is good, but I'm not sure it tackles the actual problem

Reply:
In order to solve the actual problem, I believe we may need some architectural changes. That would require `load_state_dict()`, `save_state_dict()` to be wrapped or we may need our own version of the `torch.nn.Sequential()` container. Furthermore, this solution does not cover the `Squeeze` classes either.

Reply:
that's very true